src/Index.php

```diff
@@ -25,29 +25,38 @@ class Index
     protected $logger;
 
     /**
-     * @param array $hosts
-     * @param string $name
-     * @param array $indexConfig
-     * @param SerializerInterface $serializer
+     * @param array     $hosts          only useful if a client is not provided
+     * @param string    $name           name of the main index
+     * @param array     $indexConfig    config of the index
+     * @param SerializerInterface   $serializer only useful if a client is not provided
+     * @param LoggerInterface   $logger logs connection errors
+     * @param Client   $client  If provided, the client on which every operations will be executed
+     *
+     * @deprecated The default constructor without client is deprecated since 6.5 and is replaced by "createWithoutClient", it will be removed in 7.0. Prefer using "createWithClient", which will be the default constructor in 7.0
+     *
      */
     public function __construct(
```

Comment:
Given that there are static methods dedicated to Index construction, what do you think of making this constructor private.

This would introduce a BC break but in my opinion, the ability to provide a client could be a change important enough to be considered as major.

I'd like your point of view about that.

Reply:
That pretty much was my intention with the deprecations. 

I was thinking of replacing the default constructor, by this new one instead, and only keeping the createWIthoutClient method for backwards compatibilty, hence pushing the createWithClient to the front. 

The constructor would introduce a major BC break, but at least we would have a clear constructor for doing things.

Introducing a major would make sense, but we would stop aligning with the ElasticSearch releases. Which we only kinda do right now anyway, and prevents us from doing Major versions.

So maybe it isn't that bad an idea